src/Tasks/AlgoliaIndexTask.php

```diff
@@ -222,7 +228,12 @@ private function syncCreatedPagesWithPageAlgoliaObjectIDHolders($pages) {
     }
 
     /**
-     * SynChanges removes old pages, updates pages and adds new pages to Algolia.
+     * Sync only pages with changes since the last sync, removed pages and added pages
+     *
+     * When a sync has been done we have a reference point and now can sync only the changes
```

Comment:
When a full sync has been done, we have a reference point and from then on we can sync only the changes.